vhdbuilder/packer/test-scan-and-cleanup.sh

```diff
@@ -36,7 +36,7 @@ SIG_VERSION=$(az sig image-version show \
 
 if [ -z "${SIG_VERSION}" ]; then
   echo -e "\nBuild step did not produce an image version. Running cleanup and then exiting."
-  retrycmd_if_failure 2 3 "${SCRIPT_ARRAY[@]}" || exit $?
+  retrycmd_if_failure 2 3 "${SCRIPT_ARRAY[@]}"; exit $?
```

Comment:
could we just have `exit $?` on the proceeding line since that's what we do in most other places?